pkg/spec3/path.go

```diff
@@ -74,6 +80,59 @@ func (p *Paths) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (p *Paths) UnmarshalNextJSON(opts jsonv2.UnmarshalOptions, dec *jsonv2.Decoder) error {
+	tok, err := dec.ReadToken()
+	if err != nil {
+		return err
+	}
+	var ext any
+	switch k := tok.Kind(); k {
+	case 'n':
+		return nil // noop
```

Comment:
We probably had the same thing for v2, but are we supposed to leave `p` unaffected when we get a null value, or should we reset it? That's relevant when we re-deserialize into an existing non-empty struct. jsonv2 has stronger semantics than v1 that this should behave like a json-merge-patch.

From https://pkg.go.dev/github.com/go-json-experiment/json#UnmarshalOptions.UnmarshalNext:
> In general, unmarshaling follows merge semantics (similar to [RFC 7396](https://rfc-editor.org/rfc/rfc7396.html)) where the decoded Go value replaces the destination value for any JSON kind other than an object. For JSON objects, the input object is merged into the destination value where matching object members recursively apply merge semantics.

Reply:
cc @alexzielenski, were you able to exercise this particular case in OpenAPI V2? I couldn't get a JSON example where this code path is exercised, setting "paths" to null in the JSON seems to bypass calling the UnmarshalNextJSON in this struct.

Reply:
If you unmarshal directly into a `spec.Paths` struct with JSON `null` it will hit this line.

Thats a good point Antoine. I think we do want to reset `p`

Reply:
Added resets for spec3.Paths and spec3.Responses. Should we make this change for the v2 unmarshaller as well?

Reply:
In a separate PR yes, it's relatively minor-ish (we never merge openapi specs that way ...) but we should do it for correctness.